src/redis-benchmark.c

```diff
@@ -401,8 +402,13 @@ static void randomizeClientKey(client c) {
     for (i = 0; i < c->randlen; i++) {
         char *p = c->randptr[i]+11;
         size_t r = 0;
-        if (config.randomkeys_keyspacelen != 0)
-            r = random() % config.randomkeys_keyspacelen;
+        if (config.randomkeys_keyspacelen != 0) {
+            if (sizeof(size_t) == sizeof(long long)) {
+                long long rand64 = (((long long)random()) << 32 | random());
```

Comment:
We have a 64-bit RNG in Redis: `genrand64_int64()`. See https://github.com/redis/redis/blob/unstable/src/mt19937-64.h#L68

Reply:
I saw this just now and pushed an appropriate fix. Thanks for bringing it up.